### DISCUSSION TITLE
I have a version-dependent code. How to `ifdef icarus 11 versus 12?

### DISCUSSION THREAD

**Discussion by yuri-panchul** [steveicarus/iverilog#1013] [Q&A]
I have a version-dependent code. How to `ifdef icarus 11 versus 12?

**Comment by martinwhitaker** [comment]
There is no pre-defined macro for this. You can add a `-DIVERILOG_V11` or `-DIVERILOG_V12` on the `iverilog` command line.

**Comment by caryr** [comment]
If you want something more automated the code does know the version information so you could call the compiler to get the version information and after some processing pass this using defines like Martin mentioned.
